cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/analyzer/goals/ResourceDistributionGoal.java

```diff
@@ -168,20 +161,28 @@ protected void initGoalState(ClusterModel clusterModel, Set<String> excludedTopi
    */
   @Override
   protected void updateGoalState(ClusterModel clusterModel, Set<String> excludedTopics) throws AnalysisInputException {
+    Set<Integer> brokerIdsAboveBalanceUpperLimit = new HashSet<>();
+    Set<Integer> brokerIdsUnderBalanceLowerLimit = new HashSet<>();
     // Log broker Ids over balancing limit.
     // While proposals exclude the excludedTopics, the balance still considers utilization of the excludedTopic replicas.
-    if (!_brokerIdsAboveBalanceUpperLimit.isEmpty()) {
+    for (Broker broker : clusterModel.healthyBrokers()) {
+      if (!isLoadUnderBalanceUpperLimitAfterChange(clusterModel, null, broker, REMOVE)) {
+        brokerIdsAboveBalanceUpperLimit.add(broker.id());
+      }
+      if (!isLoadAboveBalanceLowerLimitAfterChange(clusterModel, null, broker, ADD)) {
```

Comment:
The issue with the use of a non-null change-type is similar to the one that I mentioned in `isLoadUnderBalanceUpperLimitAfterChange (...)` above.
-- We use this function the same way in one more place.